plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStoragePlugin.java

```diff
@@ -66,4 +66,12 @@ protected GoogleCloudStorageService createStorageService(Environment environment
             GoogleCloudStorageClientSettings.APPLICATION_NAME_SETTING,
             GoogleCloudStorageClientSettings.TOKEN_URI_SETTING);
     }
+
+    @Override
+    public boolean reinit(Settings settings) {
+        // secure settings should be readable
+        final Map<String, GoogleCloudStorageClientSettings> clientsSettings = GoogleCloudStorageClientSettings.load(settings);
+        this.storageService.updateClientsSettings(clientsSettings);
+        return true;
```

Comment:
I don't see any documentation for the reinit() method and what it is supposed to return :(

Do you think we could simplify this too and have something like:

`return storageService.refreshAndClearCache(settings);` ?

Reply:
> I don't see any documentation for the reinit() method and what it is supposed to return :(

Well... `reinit` was vague from the start. I will document/rename/remove return type in a follow up PR, before merging to master, for all plugins. I will surely add you as a reviewer.

The same goes with `storageService#updateClientSettings` the name deserves better attention, but it **should** receive `clientsSettings` not the raw `settings` assuming the `SecureSettings` are open.

So, the renamings I will address in a follow-up PR, for all plugins (EC2, S3, Azure, GCS).

Reply:
Ok, thanks.